File: Structural/Bridge/Java/Remote.java
// Remote.java

public abstract class Remote {
  protected Device device;

  public Remote(Device device) {
    this.device = device;
  }

  public void togglePower() {
    if (device.isOn()) {
      device.turnOff();
    } else {
      device.turnOn();
    }
  }

  public abstract void volumeUp();
  public abstract void volumeDown();
}

File: Structural/Bridge/Python/remote.py
# remote.py

from abc import ABC, abstractmethod
from device import Device

class Remote(ABC):
  def __init__(self, device: Device):
    self.device = device

  def toggle_power(self):
    if self.device.is_on():
      self.device.turn_off()
    else:
      self.device.turn_on()

  @abstractmethod
  def volume_up(self):
    pass

  @abstractmethod
  def volume_down(self):
    pass
